crates/red_knot_python_semantic/resources/mdtest/expression/if.md

```diff
@@ -22,3 +22,22 @@ reveal_type(1 if None else 2)  # revealed: Literal[2]
 reveal_type(1 if "" else 2)  # revealed: Literal[2]
 reveal_type(1 if 0 else 2)  # revealed: Literal[2]
 ```
+
+## Leaked Narrowing Constraint
+
+(issue #14588)
+
+The test inside an if expression should not affect the scope outside of it.
```

Comment:
micro-nit: `if` tests do not create new scopes in Python, unlike e.g. in Rust, so we have to use the vaguer term "block" here

```suggestion
The test inside an if expression should not affect code outside of the block.
```

Reply:
Thanks!!!! :)